src/orange/orangeinp/detail/DeMorganSimplifier.cc

```diff
@@ -94,6 +94,20 @@ CsgTree DeMorganSimplifier::operator()()
     return this->build_simplified_tree();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Find the real node id to access by dereferencing Aliased nodes.
+ */
+NodeId DeMorganSimplifier::dealias(NodeId node_id) const
+{
+    NodeId dealiased{node_id};
+    while (auto const* aliased = std::get_if<Aliased>(&tree_[dealiased]))
+    {
+        dealiased = aliased->node;
+    }
+    return dealiased;
```

Comment:
Add some assertions:
```suggestion
{
    CELER_EXPECT(node_id < tree->size())
    NodeId dealiased{node_id};
    while (auto const* aliased = std::get_if<Aliased>(&tree_[dealiased]))
    {
        dealiased = aliased->node;
        CELER_ASSERT(dealiased < tree->size())
    }
    return dealiased;
```